src/components/homepage/UpEvents.astro

```diff
@@ -0,0 +1,79 @@
+---
+import { loadConfig } from "../../utils/loadconfig";
+import { EventData } from "../../utils/types";
+
+const config = loadConfig();
+const upcoming_event_link = config.events.upcoming.link;
+
+let eventsData: EventData[];
+try {
+  const response = await fetch(upcoming_event_link);
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
```

Comment:
Instead, we should log the error and return from the try block, right? Since, the try block was to make it safe from exception.

And, when we are unable to fetch the events showing something like **_"No events. Keep your eyes on for updates..."_** would be wiser move.